### PR TITLE
[decoupled-execution] retry sending commit vote

### PR DESCRIPTION
**Pull request by yuxiamit**
## Motivation

(Write your motivation for proposed changes here.)

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

## Test Plan

(Share your test plan here. If you changed code, please provide us with clear instructions for verifying that your changes work.)

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/diem/diem/tree/main/developers.diem.com, and link to your PR here.)

## If targeting a release branch, please fill the below out as well

 * Justification and breaking nature (who does it affect? validators, full nodes, tooling, operators, AOS, etc.)
 * Comprehensive test results that demonstrate the fix working and not breaking existing workflows.
 * Why we must have it for V1 launch.
 * What workarounds and alternative we have if we do not push the PR.

### CODE REVIEW

File: consensus/src/experimental/buffer_manager.rs [hunk 1]
```diff
@@ -524,9 +526,45 @@ impl StateManager {
...
+    async fn retry_broadcasting_commit_votes(&mut self) -> anyhow::Result<()> {
+        let mut cursor = self.buffer.head.clone();
+        while cursor.is_some() && !link_eq(&cursor, &self.signing_root) {
+            // we move forward before sending the message
+            // just in case the buffer becomes empty during await.
+            let next_cursor = get_next(&cursor);
+            {
+                let buffer_item = get_elem(&cursor);
+                match &*buffer_item {
+                    BufferItem::Aggregated(aggregated) => {
```
Comment (zekun000) [inline consensus/src/experimental/buffer_manager.rs:540]: why we retry on already aggregated items?

Comment (yuxiamit) [reply consensus/src/experimental/buffer_manager.rs:540]: to help other nodes

Comment (zekun000) [reply consensus/src/experimental/buffer_manager.rs:540]: that's not how we do it, we broadcast sync info to help other nodes

Comment (yuxiamit) [reply consensus/src/experimental/buffer_manager.rs:540]: But that will trigger the sync process (expensive?). It is possible that many nodes just needs one more signature to proceed. So I think broadcasting the in-buffer aggregated items can improve the situation. This is demonstrated in the previous cluster experiments that letting the commit vote carry more signatures significantly stabilizes the throughput.

Comment (zekun000) [reply consensus/src/experimental/buffer_manager.rs:540]: sync process is not supposed to be expensive or complicated, (that's one of the thing this refactor is supposed to fix). the node should just forward the aggregated LI as part of the sync request and the node could properly pop the buffer.

Comment (yuxiamit) [reply consensus/src/experimental/buffer_manager.rs:540]: A sync process will get least involve 3 network messages (sync req, fetch req, fetch resp, and potentially chunk req, chunk resp), possible txn replay, and extra disk I/O; but blindly broadcasting aggregated items will waste network resource. Maybe we can deterministically select a single node to broadcast the aggregated items? 

I'll change it to signed items for now. Later we can collect some stats.

**Review by zekun000 — DISMISSED** [review]

**Pull request closed without merging**